src/btree/bt_cursor.c

```diff
@@ -628,33 +628,37 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
 {
     WT_BTREE *btree;
     WT_CURSOR_BTREE *cbt;
+    WT_DECL_RET;
     WT_UPDATE *upd;
 
-    *updp = NULL;
-
+    *updp = upd = NULL; /* -Wuninitialized */
     cbt = (WT_CURSOR_BTREE *)cursor;
     btree = CUR2BT(cbt);
-    upd = NULL; /* -Wuninitialized */
 
     /*
-     * Not calling the cursor initialization functions, we don't want to be tapped for eviction nor
-     * do we want other standard cursor semantics like snapshots, just discard the hazard pointer
-     * from the last operation. This also depends on the fact we're not setting the cursor's active
-     * flag, this is really a special chunk of code and not to be modified without careful thought.
+     * Set the key only flag to indicate to the search that we don't want to check visibility we
+     * just want to position on a key. This short circuits validity checking.
      */
-    WT_RET(__cursor_reset(cbt));
-
-    WT_RET(btree->type == BTREE_ROW ? __cursor_row_search(cbt, false, NULL, NULL) :
-                                      __cursor_col_search(cbt, NULL, NULL));
-
+    F_SET(&cbt->iface, WT_CURSTD_KEY_ONLY);
+    ret = __wt_btcur_search(cbt);
```

Comment:
Can you add a comment here to indicate we will search the pinned page first?

Reply:
6e05ab85e